src/persist-client/src/internal/metrics.rs

```diff
@@ -1073,6 +1102,30 @@ impl CodecMetrics {
     }
 }
 
+#[derive(Debug)]
+pub struct StructuredColumnarMetrics {
```

Comment:
Can we sneak in a quick metric to cover how much time we spend doing validation plz?

Reply:
I added a metric that tracks decoding time, which is the vast majority of the time spent in validation. After we decode all we do is check if the two values are equal. More than happy to add more metrics if need be!